html/pfappserver/root/static.alt/src/components/pfCSVImport.vue

```diff
@@ -0,0 +1,1145 @@
+                :value="page"
+                :total-rows="pageMax * perPage"
+                :per-page="perPage"
+                :disabled="isDisabled"
+                @change="setPage($event)"
+              >
+                <template v-slot:page="{ page, active }">
+                  <b v-if="active">{{ page }}</b>
+                  <i v-else>{{ page }}</i>
+                </template>
+              </b-pagination>
+            </b-col>
+          </b-row>
+          <!-- table -->
+          <div class="pf-csv-import-table" :class="{ 'hover': hover, 'striped': striped }">
+            <!-- table head -->
+            <b-row class="pf-csv-import-table-head">
+              <b-col class="text-nowrap">
+                {{ $t('Field Mappings') }}
+              </b-col>
+              <template v-for="(_, colIndex) in new Array(perPage)">
+                <b-col class="text-nowrap" :key="colIndex">
+                  <template v-if="((perPage * page) - perPage + colIndex + 1) <= (linesCount - ((config.header) ? 1 : 0))">
+                    {{ $t('Line') }} #{{ (perPage * page) - perPage + colIndex + 1 }}
+                  </template>
+                  <template v-else>
+                    <icon name="ban" class="text-secondary"/>
+                  </template>
+                </b-col>
+              </template>
+            </b-row>
+            <!-- table body -->
+            <b-row class="pf-csv-import-table-row" v-for="(_, rowIndex) in previewColumnCount" :key="rowIndex">
+              <b-col>
+                <b-form-group
+                  :state="($v && 'importMapping' in $v && $v.importMapping.$invalid) ? false : null"
+                  :invalid-feedback="getImportMappingVuelidateFeedback()"
+                  class="my-1 pf-csv-import-form-group"
+                >
+                  <b-input-group>
+                    <template v-slot:append v-if="rowVariant(rowIndex)">
+                      <b-button variant="light" class="pb-1" :class="`text-${rowVariant(rowIndex)}`" :disabled="isDisabled" @click="deleteImportMapping(rowIndex)">
+                        <icon name="times-circle"/>
+                      </b-button>
+                    </template>
+                    <b-form-select
+                      v-model="importMapping[rowIndex]"
+                      :disabled="isDisabled"
+                    >
+                      <template v-slot:first>
+                        <optgroup :label="$t('Ignored fields')">
+                          <option :value="null" class="bg-danger text-white">{{ $t('Ignore') }}</option>
+                        </optgroup>
+                      </template>
+                      <optgroup :label="$t('Required fields')">
+                        <option v-for="option in importMappingOptions.filter(o => o.required)" :key="option.value" :value="option.value" :disabled="option.disabled" :class="{'bg-success text-white': !option.disabled}">{{ option.text }}</option>
```

Comment:
The styling/classes on `option` has no effect, at least with Chrome. We should probably remove it or use a `pfFormChosen`.

Reply:
Styling works with Chromium Version 71.0.3578.98 (Official Build) (64-bit)

![Administrator - PacketFence - Google Chrome_066](https://user-images.githubusercontent.com/3904468/67959256-d1034d00-fbce-11e9-8d2a-1dd10d9a8677.png)

Reply:
I think you may be referring to the outer <select> tag, which can't be styled. Correct me if I'm wrong.

Reply:
Be aware that changing the background colour of a `option` element is not possible under macOS nor Windows. I would suggest to use a `pfFormChosen` component.